core/services/job/orm.go

```diff
@@ -195,7 +195,9 @@ func (o *orm) CreateJob(ctx context.Context, jobSpec *Job, p pipeline.Pipeline)
 	// Inherit the parent context so that client side request cancellations are respected.
 	ctx, cancel := context.WithTimeout(ctx, postgres.DefaultQueryTimeout)
 	defer cancel()
-	return postgres.GormTransaction(ctx, o.db, func(tx *gorm.DB) error {
+	outerTx := postgres.TxFromContext(ctx, o.db)
```

Comment:
This is taking the transaction which was injected into the context by the caller. This is used by the `postgres.GormTransaction` below to nest it's own transaction within the caller's transaction. The inner transaction will rollback the outer transaction if it errors

https://gorm.io/docs/transactions.html#Nested-Transactions

Reply:
Postgres does not support nested transactions, I imagine this emulates them via savepoint. We probably don't want to do that.

Reply:
OK I can change the callers of this to use the transaction wrapper instead then